core/src/main/java/org/apache/calcite/sql/SqlIntervalQualifier.java

```diff
@@ -584,6 +585,41 @@ private int[] evaluateIntervalLiteralAsMonth(
     }
   }
 
+  /**
+   * Validates an INTERVAL literal against a WEEK interval qualifier.
+   *
+   * @throws org.apache.calcite.runtime.CalciteContextException if the interval
+   * value is illegal
+   */
+  private int[] evaluateIntervalLiteralAsWeek(
```

Comment:
This looks correct based on the other similar functions, but I'm still not fully understanding what this is doing.

Reply:
Me neither :) I should probably rectify that...